sdk/test-utils/recorder/src/matcher.ts

```diff
@@ -5,13 +5,41 @@ import { createPipelineRequest, HttpClient } from "@azure/core-rest-pipeline";
 import { paths } from "./utils/paths";
 import { RecorderError } from "./utils/utils";
 
-export type Matcher = "HeaderlessMatcher" | "BodilessMatcher";
+export type Matcher = "HeaderlessMatcher" | "BodilessMatcher" | "CustomDefaultMatcher";
+
+/**
+ * Body the customer matcher expects.
+ */
+export interface CustomMatcherBody {
+  /**
+   * Should the body value be compared during lookup operations?
+   */
+  compareBodies?: boolean;
+  /**
+   * Array of additional headers that should be excluded during matching.
+   * "Excluded" headers are entirely ignored. Unlike "ignored" headers, the presence (or lack of presence) of a header will not cause mismatch.
+   */
+  excludedHeaders?: string[];
+  /**
+   * Array of additional headers that should be ignored during matching.
+   * Any headers that are "ignored" will not do value comparison when matching.
+   * This means that if the recording has a header that isn't in the request, a test mismatch exception will be thrown noting the lack of header in the request.
+   *
+   * This also applies if the header is present in the request but not recording.
+   */
+  ignoredHeaders?: string[];
```

Comment:
Just wondering - is there a usecase for this? If yes, awesome! Otherwise (feel free to ignore if that's already the case of course) I would advocate leaving this out and adding it only if someone needs it. What do you think?

My only reasoning is that the difference between ignoredHeaders and excludedHeaders seems subtle and can be confusing which one to choose...

Reply:
I realize these are probably already defined in the test proxy and we're just making them available here is that right? I only assume that because I don't see them get used anywhere. I still think we can expose a subset of these options until we need all of them (unless we need all of them today)...

Reply:
I was wondering the same.
That's a good idea actually, I'll take your suggestion.